### PR TITLE
support minLength, maxLength with Array

### PR DESCRIPTION
**Pull request by jiho-kr**
support minLength, maxLength with validate.Array

validate.Array([validate.ObjectId], {minLength: 1, maxLength: 50})

http://git.sphd.io/support/stt-issues/issues/388

new Schema Type html (require donghlee)

RELEASE_CANDIDATE

### CODE REVIEW

**Comment by kuyh** [comment]
approved

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "island",
-  "version": "1.10.0-dev",
+  "version": "1.10.1-dev",
```
Comment (heycalmdown) [inline package.json]: no need to bump the version up at master

File: src/controllers/endpoint-decorator.ts [hunk 2]
```diff
@@ -397,23 +400,34 @@ export namespace validate {
...
+  }
+
+  // tslint:disable-next-line class-name
   export class _Array {
     items: [ValidatePropertyTypes] | undefined;
+    minLength?: number;
+    maxLength?: number;
 
-    constructor(items: [ValidatePropertyTypes] | undefined) {
+    constructor(items: [ValidatePropertyTypes] | undefined, opts?: __Array | undefined) {
```
Comment (heycalmdown) [inline src/controllers/endpoint-decorator.ts]: The `?` means that the argument can be undefined. So the later `undefined` will be redundant.

File: src/spec/schema-decorator.spec.ts [hunk 3]
```diff
@@ -231,6 +231,40 @@ describe('validate', () => {
...
+
+  it('should support array options - minLength', () => {
+    const result = v.validate(v.Array([Number], {minLength: 5 }));
+    expect(result).toEqual({
+      type: 'array',
+      items: {type : 'number',
+              minLength: 5,
+              optional: false
+             }
+    });
```
Comment (heycalmdown) [inline src/spec/schema-decorator.spec.ts]: The expected result should look like 

```ts
expect(result).toEqual({
  type: 'array',
  minLength: 5,
  optional: false,
  items: {type: 'number', optional: false}
});
```

Check [this](https://github.com/Atinux/schema-inspector/blob/master/README.md#v_length)

File: src/controllers/endpoint-decorator.ts [hunk 4]
```diff
@@ -319,6 +319,9 @@ export namespace validate {
   // tslint:disable-next-line class-name
   export interface _Any { $validate: Symbol; };
   export const Any: _Any = { $validate: Symbol() };
+  // tslint:disable-next-line class-name
+  export interface _Html { $validate: Symbol; };
+  export const Html: _Html = { $validate: Symbol() };
```
Comment (heycalmdown) [inline src/controllers/endpoint-decorator.ts]: Can you provide a test for this? (see [schema-types.spec.ts](https://github.com/spearhead-ea/island/blob/4ce37d0af67acab3c0af9462f3d21d57a8b97842/src/spec/schema-types.spec.ts))

And, you are missing to add a piece of code into [schema-types.ts](https://github.com/spearhead-ea/island/blob/4ce37d0af67acab3c0af9462f3d21d57a8b97842/src/middleware/schema-types.ts)

File: src/spec/schema-decorator.spec.ts (hunk 3 above)
Comment (jiho-kr) [inline src/spec/schema-decorator.spec.ts]: It differs from the example of schema-inspector.
like this at schema-inspector.
```
var inspector = require('schema-inspector');

var schema = {
	type: 'array',
	items: { type: 'string', minLength: 8, maxLength: 11 }
};
```

Comment (jiho-kr) [inline src/spec/schema-decorator.spec.ts]: ok, thanks :) I had misunderstood.

**Comment by heycalmdown** [comment]
close #52

**Pull request merged**